Content: Node port

SallyMcGrath opened the pull request:
## What does this change?

Module: Node
Week(s): 1-4

Addressing #352 

## Checklist

- [x] I have read the [contributing guidelines](CONTRIBUTING.MD)
- [x] I have checked my spelling and grammar with an [automated tool](https://www.grammarly.com/grammar-check)
- [x] I have previewed my changes to check the [markdown renders](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax) as I intend
- [x] I have run my code to check it works
- [x] My changes follow our [Style Guide](https://curriculum.codeyourfuture.io/guides/code-style-guide)

## Description

Addressing #352 
Storing my port of Node from syllabus
- [x] Sprint 1 Prep
- [x] Sprint 1 Dayplan
- [x] Sprint 2 Prep
- [x] Sprint 2 Dayplan
- [x] Sprint 3 Prep
- [x] Sprint 3 Dayplan
- [x] Sprint 4 Prep
- [x] Sprint 4 Dayplan

Will be draft until I've filled it all in

Notes

Not rewriting this but making tiny upgrades as I go if it's as easy as not:

I made a runkit render hook so they can run a server on the page
I replaced a picture of a url diagram with a diagram
I dumped the starter kit and have them make their own as the instructions do seem to expect)

## Who needs to know about this?

<!-- Tag anyone who might want to be notified about this PR -->

SallyMcGrath commented:
I cannot merge this until the workshops are merged

https://github.com/CodeYourFuture/CYF-Workshops/pulls?q=is%3Apr+is%3Aopen+label%3A%22New+Workshop%22

SallyMcGrath commented:
Then I will update the dayplans with the right links. Right now they have placeholders so the build doesn't fail

40thieves commented on content/en/servers/_index.md:
```diff
@@ -0,0 +1,8 @@
++++
+title = 'Servers'
+description = 'Write server-side JavaScript with Node.js; develop CRUD applications with Express.js; write and test APIS with Postman'
```
```suggestion
description = 'Write server-side JavaScript with Node.js; develop CRUD applications with Express.js; write and test APIs with Postman'
```

40thieves commented on content/en/servers/prep/index.md:
```diff
@@ -0,0 +1,15 @@
+backlog= 'Module-Node'
+[[blocks]]
+name="Install Postman"
+src="node/blocks/postman"
```
```suggestion
src="servers/blocks/postman"
```

40thieves commented on content/en/servers/sprints/1/prep/index.md:
```diff
@@ -0,0 +1,36 @@
+emoji= '📝'
+menu_level = ['sprint']
+weight = 1
+backlog= 'Module-Node'
```
Probably one for follow up, but if we're happy with the Servers naming, then we should probably rename this too.

40thieves reviewed (changes requested):
Looks like from a quick spot-check that most of the block paths haven't been updated from `node` to `servers` (to match the file renames), so this is quite broken at the moment.

I'll see if I can pull this down and do a quick find/replace

40thieves commented:
Ok I think I've fixed that problem but now I think it needs rebasing on https://github.com/CodeYourFuture/curriculum/pull/470, so I would do that and try it locally but I've got another build problem where the assets won't load.

So I think I'm going to call it there for now. Will try to get back to this another time.

SallyMcGrath commented:
> Ok I think I've fixed that problem but now I think it needs rebasing on #470, so I would do that and try it locally but I've got another build problem where the assets won't load.
> 
> So I think I'm going to call it there for now. Will try to get back to this another time.

Sorry about that @40thieves . Appreciate you taking the time. I've fixed the preview now.

40thieves commented:
I think that asset fix needs to land on `main` - I was stuck there too!

SallyMcGrath commented:
> I think that asset fix needs to land on `main` - I was stuck there too!
> 
> As in, I appreciate it being fixed here, but I wouldn't want to block that fix for everyone on me reviewing this.

Yep: #473 here you go

40thieves commented on content/en/servers/blocks/crud/index.md:
```diff
@@ -0,0 +1,31 @@
+
+### 🎯 Goal
+
+Our API will manage movie data. It will:
+
+✨ **C**reate a new movie,
+🐕 **R**etrieve a list of movies or a single movie
```
```suggestion
For now, we'll focus on the first two of these. We'll make a server that provides an API for managing movie data. It will:

✨ **C**reate a new movie,
🐕 **R**etrieve a list of movies or a single movie
```

40thieves commented on content/en/servers/blocks/get-single/index.md:
```diff
@@ -0,0 +1,32 @@
+
+Sometimes, we do not want to list all the information in one request, maybe we only want to get the information related to a single movie. Imagine a page to display the details of one movie. We'd call the server, get all movies, then filter the one we need client-side. It would be more effective to tell the server to just return the one movie we are interested in.
+
+We will now add a new endpoint to return only a single movie GET /movies/:movieId. In this case, movieId will tell us what movie we can return. The call will be `GET /movies/10` and that will return the movie with that has `movieId: "10"`.
```
```suggestion
We will now add a new endpoint to return only a single movie `GET /movies/:movieId`. In this case, `movieId` will tell us what movie we can return. The call will be `GET /movies/10` and that will return the movie with that has `movieId: "10"`.
```

40thieves commented on content/en/servers/blocks/crud-2/index.md:
```diff
@@ -0,0 +1,30 @@
+
+### 🎯 Goal
+
+Our API will manage movie data. It will:
+
+✨ **C**reate a new movie,
+🐕 **R**etrieve a list of movies or a single movie
+📨 **U**pdate a movie
```
```suggestion
Now we'll focus on updating resources. We'll make a server that provides an API for managing movie data. Previously we built an API that will:

✨ **C**reate a new movie,
🐕 **R**etrieve a list of movies or a single movie

Now we'll extend this to also:

📨 **U**pdate a movie
```

40thieves commented on content/en/servers/blocks/put/index.md:
```diff
@@ -0,0 +1,25 @@
+
+This means that `PUT /movies/2` should update an movie with the id `2` and return `200` with `JSON { success: true }` to the user.
+
+The code should look something like this:
```
I think we should either say this is starter code, or fill out the example below to demonstrate the actual handling of updating a movie in the snippet.

I think I'd lean towards the former:

```suggestion
To get you started, the code for handling a `PUT` request is here:
```

40thieves commented on content/en/servers/blocks/put/index.md:
````diff
@@ -0,0 +1,25 @@
+});
+```
+
+Remember, you have got to update the movie, not add it to the list.
````
I'm not sure this is a blocker for merging, but the requirements here are a bit unclear: _how_ are we updating the movie?

40thieves commented on content/en/servers/blocks/put/index.md:
```diff
@@ -0,0 +1,25 @@
+
+Remember, you have got to update the movie, not add it to the list.
+
+Test that your API works by updating one of the movies.
```
Again, almost certainly out-of-scope for now, but is it worth trying to say that the movie is updated in memory and killing the server will effectively revert it.

40thieves commented on content/en/blocks/wordle/index.md:
```diff
@@ -0,0 +1,7 @@
+time= 5
++++
+
+<iframe src="https://www.nytimes.com/games/wordle/index.html" width="100%" height="640px"></iframe>
```
I think we probably should add some context here:

```suggestion
During the class we'll build our own version of the Wordle game below. To make sure you understand the rules of Wordle, try playing below:

<iframe src="https://www.nytimes.com/games/wordle/index.html" width="100%" height="640px"></iframe>
```

40thieves commented on content/en/servers/blocks/intro-to-express/index.md:
```diff
@@ -0,0 +1,89 @@
+facilitation = false
+emoji= '🧩'
+[objectives]
+    1='Use the Teach Tech Together guide to construct your objectives'
```
I'm no good at writing these, but here's a first attempt

```suggestion
    1='Create an Express server'
```

40thieves commented on content/en/servers/blocks/render/index.md:
I'll cc @sztupy here in light of our recent discussions about switching away from Render.

40thieves reviewed (approved):
Some minor suggestions/copy-edits but happy to go ahead and merge this.

I'll try to get onto the workshops soon!

40thieves commented:
Actually I ended up having some minor copy-edits that seemed reasonable uncontroversial so I just pushed them onto the branch. Hope that's ok - feel free to revert them if you want.

SallyMcGrath commented:
> Actually I ended up having some minor copy-edits that seemed reasonable uncontroversial so I just pushed them onto the branch. Hope that's ok - feel free to revert them if you want.

Yes, please do. By the way, it's so weird getting line edits on content you haven't written. I read your emails and was like... I never wrote that, did I? No, I didn't ! It's a port! 😂
